django/librarian/tasks.py

```diff
@@ -41,17 +42,23 @@ def process_document(document_id, language=None):
     document.celery_task_id = current_task.request.id
     document.save()
 
-    with translation.override(language):
-        try:
-            process_document_helper(document)
+    llm = OttoLLM()
+    try:
+        with translation.override(language):
+            process_document_helper(document, llm)
+
+    except SoftTimeLimitExceeded:
+        document.status = "ERROR"
+        document.celery_task_id = None
+        document.save()
 
-        except SoftTimeLimitExceeded:
-            document.status = "ERROR"
-            document.celery_task_id = None
-            document.save()
+    user = User.objects.get(upn=user_upn) if user_upn else None
+    costs = llm.create_costs(user, feature="qa", embed_task="doc")
+    document.cost = costs[0].usd_cost
+    document.save()
 
 
-def process_document_helper(document):
+def process_document_helper(document, llm):
```

Comment:
Again, "llm" is a slight misnomer, but this is the OttoLLM object that includes the vector store helpers and costing stuff.